flow/model/qvalue/qvalue_converter.go

```diff
@@ -0,0 +1,25 @@
+package qvalue
+
+import "github.com/shopspring/decimal"
+
+type TypeConversion struct {
```

Comment:
this can be generic over `TypeConversion[TFrom, TTo]` then implement an interface which takes `Convert(from QValue) QValue` that handles downcasting. That way your methods can take concrete QValue types instead of having to call `.Value()` internally

Reply:
discussed this via DM, will handle this separately later